### PR TITLE
PR for terraform tunnel example with GCP

### PR DESCRIPTION
**Pull request by tenaciousdlg**
Adding in a new directory to the examples dir for terraform usage of tunnels

### CODE REVIEW

**Review by chungthuang — APPROVED** [review]

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+This demo creates a Google Cloud (GCP) instance running an httpbin docker container.
+
+It also spins up a Cloudflare Tunnel on the GCP instance proxying traffic to the httpbin container
+and the local SSH port on the server.
+
+DNS records are created at Cloudflare to point to the tunnel endpoint. 
+
+The GCP instance is set to not accept SSH traffic via a `no-ssh` tag. Your Firewall settings may differ and this may need to be adjusted as needed.
+
+An access policy is create for `ssh.yourdomain.com` and the only person allowed is the user
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: create -> created

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+It also spins up a Cloudflare Tunnel on the GCP instance proxying traffic to the httpbin container
+and the local SSH port on the server.
+
+DNS records are created at Cloudflare to point to the tunnel endpoint. 
+
+The GCP instance is set to not accept SSH traffic via a `no-ssh` tag. Your Firewall settings may differ and this may need to be adjusted as needed.
+
+An access policy is create for `ssh.yourdomain.com` and the only person allowed is the user 
+provided to `cloudflare_email` in `terraform.tfvars`.
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: maybe change that variable to `access_email`?

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Update your workstation's SSH config to provide a specific host block for `ssh.yourdomain.com`.
+
+Here is an example:
+
+```
+Host ssh.atxflare.cf
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: maybe switch `ssh.atxflare.cf` with `ssh.yourdomain.com` for coherency

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    IdentityFile /Users/cdlg/.ssh/google_compute_engine
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: I'd replace `cdlg` with `user` (to map the agnostic example below with the ssh command)

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    ProxyCommand /usr/local/bin/cloudflared access ssh --hostname %h
+```
+
+You can then `ssh user@ssh.yourdomain.com` where user is your OS user to login.
+
+## Endpoints
+
+> Where `yourdomain.com` is the zone assinged to `cloudflare_zone` in `terraform.tfvars`
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: Is this `>` a typo?

Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: assinged -> assigned

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+`yourdomain.com`
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: This `## Endpoints` section would be a bit more readable if it was a list, no?

```
## Endpoints
 - `yourdomain.com` allows access to the HTTPBin webserver
 - `ssh.yourdomain.com` allows access to the SSH server running on the GCP instance
```

File: terraform-zerotrust-ssh-http-gcp/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+HTTPBin container 
+
+`ssh.yourdomain.com`
+
+SSH end point of the GCP compute instance
+
+# Terraform version
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/README.md]: Maybe create a `#Requirements` section on the top listing necessary software/versions?

File: terraform-zerotrust-ssh-http-gcp/argo.tf [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# The random_id resource is used to generate a 35 character secret for the tunnel
+resource "random_id" "argo_secret" {
```
Comment (nmldiegues) [inline terraform-zerotrust-ssh-http-gcp/argo.tf]: I'd refer to it as `tunnel_secret` (as cloudflared logs do) rather than `argo_secret`

**Review by nmldiegues — COMMENTED** [review]
Awesome job!

My remarks are all cosmetic. Feel free to not follow them as you please. Just let me know when to re-review.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 1 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Change completed.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 2 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: My only hesitation here is that the email is also associated with their Cloudflare account and is used to auth and deploy Cloudflare resources. Perhaps a (i.e. the email you'll use to login via Access) after the mention of `cloudflare_email`?

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 3 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Change completed.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 4 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Change completed.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 5 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Ah good catch, I think I originally wanted this as an inline comment but it looks better without the formatting. Resolved.

Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Change completed.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 6 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Agreed and change completed.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 7 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Good idea, I switched the README around to reflect that

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 2 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Added an update for this. Let me know if it resonates or if another change is needed.

File: terraform-zerotrust-ssh-http-gcp/argo.tf (hunk 8 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/argo.tf]: Good call and I updated this on both resources in `argo.tf` and the reference to the resource in `instance.tf`.

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 2 above)
Comment (nmldiegues) [reply terraform-zerotrust-ssh-http-gcp/README.md]: LGTM

**Review by nmldiegues — APPROVED** [review]

File: terraform-zerotrust-ssh-http-gcp/README.md (hunk 2 above)
Comment (tenaciousdlg) [reply terraform-zerotrust-ssh-http-gcp/README.md]: Fantastic, thanks!

**Pull request merged**